src/flb_engine.c

```diff
@@ -1135,6 +1135,12 @@ int flb_engine_shutdown(struct flb_config *config)
         flb_hs_destroy(config->http_ctx);
     }
 #endif
+    if (config->evl) {
```

Comment:
it seems to me that before restart the event loop is not being destroyed, I would suggest to identify the root cause for this

Reply:
I will look into this. This specific piece of code was added in to mitigate crashes in the following tests:

- flb-rt-config_map_opts
- flb-rt-custom_calyptia_test
- flb-rt-filter_stdout
  - case_insensitive_name

The other `flb-rt-filter_stdout` test, `json_multiple`, does not crash in the same manner as the other test.

This is most likely due to the manner in which these tests use the fluent-bit API to instantiate the pipelines they use. The `case_insensitive_name` test executes `flb_destroy` without first calling `flb_stop`. Upon trying to call `flb_stop` in the test it crashes when attempting to call `pthread_join` inside `flb_stop`. This is most likely due to the fact that the test also does not call `flb_start` to initialize the thread.

The easiest way out of this would be to simply leave the check there. Another alternative would be to move the deletion of the event channel into flb_stop, where it might be better placed.

Reply:
Moving `config->is_running` into `flb_engine_started` avoids the SIGSEGV caused by destroying the `ch_self_events` channel but causes the memory used by all the custom, input, output and filter plugins to be leaked when using in several tests. The deallocation of plugins could be moved to `flb_config_exit` or similar, but that seems to me be a bit ouf of scope.